Rpackage/R/bestsplit.R

```diff
@@ -0,0 +1,64 @@
+bestsplit <- structure(function
+### Compute vector of optimal prediction and cost.
+#We have the feature and target value of the data that is to be splitted.
+(target.mat, feature.mat, margin=0.0, loss="hinge"){
+  #We keep track of the following values for each optimal split
+  best_split <- NULL
+  best_split$cost <- Inf
+  best_split$br <- NULL
+  best_split$varid <- NULL
+  best_split$row <- NULL
+
+  #loop for every feature
+  for (index in 2:(length(feature.mat[1,]))){
```

Comment:
Why do you start at index 2?

Reply:
the 1st index is the target...in mmit (line 6,7) I defined the 1st column to be the target

Reply:
Ok. It makes sense then. But it seems like an odd choice, since you also take target.mat as an argument. Why did you choose to have the 1st column be the target? Wouldn't it be more logic to have only features in feature.mat?

Reply:
True...but didn't think of that while making it...it seems redundant to pass target as well as add it in dataframe....shall I change it?